core/shared/src/main/scala/zio/Cancelable.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+trait Cancelable[+E, +A] {
```

Comment:
This is only used in `Cancelable`, right? In that case, I'd inline it completely so we can delete the class.

Reply:
@jdegoes You were actually correct, there is `java.util.Cancelable`

```scala
def cancel(mayInterruptIfRunning: Boolean): Boolean
def isCancelled(): Boolean
def isDone(): Boolean
```

Let’s use that.

Reply:
That JCP was never accepted,